services/movie_session.py

```diff
@@ -0,0 +1,51 @@
+from django.db.models import QuerySet
+
+from db.models import MovieSession, CinemaHall, Movie
+
+import datetime
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int) -> Movie:
+    movie = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
+        movie=Movie.objects.get(id=movie_id)
+    )
+
+    return movie
+
+
+def get_movies_sessions(session_date: datetime = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+        return queryset
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: datetime = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> None:
+    movie_session = MovieSession.objects.filter(id=session_id)
+
+    if show_time:

```

Comment:
If you are trying to reduce the number of queries then you should use Python object interface instead of `.update()` method and call `.save()` at the end of the function.

Reply:
okaay, i'll implement it. thank u a lot!